packages/wrangler/src/dev.tsx

```diff
@@ -930,7 +958,7 @@ export async function startDev(args: StartDevOptions) {
 					minify={args.minify ?? configParam.minify}
 					nodejsCompatMode={nodejsCompatMode}
 					build={configParam.build || {}}
-					define={{ ...configParam.define, ...cliDefines }}
+					define={{ ...processEnvValues, ...configParam.define, ...cliDefines }}
```

Comment:
This also needs to be added to the`--x-dev-env` codepath

https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/api/startDevWorker/ConfigController.ts#L250

Reply:
thanks! updated the PR with support for `--x-dev-env`, and e2e tests